src/html-templates.ts

```diff
@@ -53,9 +60,13 @@ export function getTemplate(
   >
     ${slotsTemplate}${slot || ""}
 </${unsafeStatic(component!.tagName!)}>
-<script>
-  component = document.querySelector('${component!.tagName!}');
-</script>
+${
+  options.hideScriptTag
```

Comment:
Thios will still need to display if any component properties are set.

Reply:
@break-stuff - what are your feelings on making this default to true? I would imagine that in most cases the story isn't going to need to access 'component', in my experience it is usually only in exceptional circumstances that you need to do this sort of thing.

Reply:
If we did that, it would introduce a breaking change. That 's something we can discuss with v2, but I think it should stay consistent with the existing behavior for now.

Reply:
@break-stuff It's not clear to me why the script tag still needs to display when any component properties are set. The properties are also set on the component using lit-attributes right? With the SpreadDirective? I'm not very well-known with lit, but what would the component reference still be needed for?

Reply:
@ArcoMul - its sometimes useful to be able to do this sort of thing if you want to demo calling methods on the component, I've done this in the past but (as mentioned above) it isn't something most people will need much of the time. To me it feels like 'component' is a helper step too far, if a consumer needs to access the component from a script then they can always get it via document.querySelector.

Reply:
@break-stuff I'm wondering if the helper could treat the extra styles/scripts as storybook decorators, this flag could then be used to determine whether or not to include them? https://storybook.js.org/docs/api/doc-block-source#excludedecorators

Reply:
@ArcoMul, you are correct. The spread operator should cover that. I was recalling an earlier implementation.

@rstoneIDBS, we could probably do that for the scripts, but the styles need to be there for CSS Parts and Properties binding.